pkg/zkcertificate/certificate_uniswap.go

```diff
@@ -0,0 +1,105 @@
+// Copyright © 2024 Galactica Network
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package zkcertificate
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+
+	"github.com/iden3/go-iden3-crypto/poseidon"
+	"github.com/shopspring/decimal"
+
+	"github.com/galactica-corp/guardians-sdk/internal/validation"
+)
+
+// UniswapInputs represent the input data for the Uniswap verification.
+type UniswapInputs struct {
+	Address            string          `json:"address" validate:"required,eth_addr"`
```

Comment:
My tests fail somehow with the address param. Am I doing something wrong?

- If I use `"address": "0xDf10AB6DaF3476a45A731a32F0FA434679E90761"`, I get `Error: read certificate content: encode inputs to finite field: invalid address: encoding/hex: invalid byte: U+0078 'x'`.
- If I use `"address": "Df10AB6DaF3476a45A731a32F0FA434679E90761",`, I get `Error: read certificate content: read uniswap inputs: decode json: Key: 'UniswapInputs.Address' Error:Field validation for 'Address' failed on the 'eth_addr' tag`.

Reply:
I overlooked, thanks) Fixed in 03812cf